src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFilteringProfile.cpp

```diff
@@ -109,9 +109,9 @@ namespace OpenMS
     }
     
     // spline interpolate the profile data
-    for (MSExperiment::Iterator it = exp_profile.begin(); it < exp_profile.end(); ++it)
+    for (auto& profile : exp_profile)
```

Comment:
e.g., here it is not clear that we are dealing with a spectrum by replacing it with auto.
also it can be const.

better would be for (const MSSpectrum& spec : exp_profile)